packages/menu-bar/src/vaadin-menu-bar-button.js

```diff
@@ -37,6 +37,20 @@ class MenuBarButton extends Button {
   static get is() {
     return 'vaadin-menu-bar-button';
   }
+
+  /**
+   * Override method inherited from `ButtonMixin`. A Space or an Enter key press should not result in a button click for a menu bar button. It also has to focus on the first item in the submenu. These cases are handled in `VaadinMenuBarMixin`.
+   *
+   * @param {KeyboardEvent} event
+   * @protected
+   * @override
+   */
+  _onKeyDown(event) {
+    if (this._activeKeys.includes(event.key)) {
+      return;
+    }
```

Comment:
We definitely want to fire a click event on Space or Enter. Otherwise it would not possible anymore to trigger buttons with the keyboard. At least for buttons without a menu, but probably also for buttons with a menu.

For example:
```html
    <vaadin-menu-bar></vaadin-menu-bar>
    <script type="module">
      const menuBar = document.querySelector('vaadin-menu-bar');
      menuBar.items = [
        { text: 'Edit' },
      ];

      document.querySelectorAll('vaadin-menu-bar-button').forEach((button) => {
        button.addEventListener('click', (e) => {
          // Doesn't work anymore
          const item = e.currentTarget;
          console.log(item.textContent);
        });
      });
    </script>
```

Reply:
Well, technically there is the item selected event to listen for. The Flow component has convenience methods for adding click listeners, but those seem to be added to a nested element.

So maybe it's not a big issue after all. Still weird that a button does not trigger a click event.

Reply:
From what I understand the issue is that the method for opening the submenu is called twice:
- First from the click event that is triggered by the keydown event
- A second time from the key down event itself

Due to the click event the overlay itself is focused, and the second method call doesn't move the focus to the item. 

An alternative to preventing the click event could be to add a check whether keyboard navigation was used, which doesn't seem to be reset from triggering the click event programmatically:
```js
          // Do not focus item when open not from keyboard
          if (!keydown && !isKeyboardActive()) {
            overlay.$.overlay.focus();
          }
```

Reply:
Looks like we might even be able to remove the `keydown` parameter here if we use the `isKeyboardActive` check. The new Enter + Space handlers in the same module should also be obsolete.

Reply:
I tried the approach with `isKeyboardActive()`. However, if programmatically clicked, it doesn't always work. This is a problem when keyboard was active, and then `click()` is triggered with no user input. It does not update keyboard active state, which results in a first item focus. This can be observed in one of RTL Material tests. 

For this reason, I added a flag to menu bar button, which is updated if Enter or Space is used. Then, this flag is used in the click listener in menu bar to determine whether the click was triggered by key presses.

Reply:
Current solution looks good. Kind of curious why `isKeyboardActive` is active in the visual tests, but no need to solve this right now.